syntax = "proto3";
//python -m grpc_tools.protoc -I. --python_out=.  --grpc_python_out=. library.proto
package library;

// Service Definition
service Library {
  rpc GetBookDetails (GetBookDetailsRequest) returns (GetBookDetailsReply) {}
  rpc GetAvailableBooks (GetAvailableBooksRequest) returns (GetAvailableBooksReply) {}

}

// Message Definition
message GetBookDetailsRequest {
  int32 book_id = 1;
}

message GetBookDetailsReply {
  string book_name = 1;
  string author_name = 2;
  string description = 3;
  string condition = 4;
}
message GetAvailableBooksRequest {
	
}

message GetAvailableBooksReply {
  string book_name = 1;
}